easybuild/tools/containers/docker.py

```diff
@@ -0,0 +1,179 @@
+from easybuild.tools.run import run_cmd
+from .base import ContainerGenerator
+from .utils import det_os_deps
+
+_log = fancylogger.getLogger('tools.containers.singularity')  # pylint: disable=C0103
+
+DOCKER_UBUNTU1604_TMPL = """\
+FROM ubuntu:16.04
+LABEL maintainer=mohamed.abidi@brightcomputing.com
+
+RUN apt-get update && \\
+    apt-get install -y python python-pip lmod curl wget
+
+RUN pip install -U pip setuptools && \\
+    hash -r pip && \\
+    pip install -U easybuild
+
+RUN mkdir /app && \\
+    mkdir /scratch && \\
+    mkdir /scratch/tmp && \\
+    useradd -m -s /bin/bash easybuild && \\
+    chown easybuild:easybuild -R /app && \\
+    chown easybuild:easybuild -R /scratch
+
+RUN OS_DEPS='%(os_deps)s' && \\
+    test -n "${OS_DEPS}" && \\
+    for dep in ${OS_DEPS}; do apt-get -qq install ${dep} || true; done
+
+USER easybuild
+
+RUN set -x && \\
+    . /usr/share/lmod/lmod/init/sh && \\
+    eb %(eb_opts)s --installpath=/app/ --prefix=/scratch --tmpdir=/scratch/tmp
+
+RUN touch ${HOME}/.profile && \\
+    echo '\\n# Added by easybuild docker packaging' >> ${HOME}/.profile && \\
+    echo 'source /usr/share/lmod/lmod/init/bash' >> ${HOME}/.profile && \\
+    echo 'module use %(init_modulepath)s' >> ${HOME}/.profile && \\
+    echo 'module load %(mod_names)s' >> ${HOME}/.profile
+
+CMD ["/bin/bash", "-l"]
+"""
+
+DOCKER_CENTOS7_TMPL = """\
+FROM centos:7
+LABEL maintainer=mohamed.abidi@brightcomputing.com
+
+RUN yum install -y epel-release && \\
+    yum install -y python python-pip Lmod curl wget git
+
+RUN pip install -U pip setuptools && \\
+    hash -r pip && \\
+    pip install -U easybuild
+
+RUN OS_DEPS='%(os_deps)s' && \\
+    test -n "${OS_DEPS}" && \\
+    yum --skip-broken install -y "${OS_DEPS}" || true
+
+RUN mkdir /app && \\
+    mkdir /scratch && \\
+    mkdir /scratch/tmp && \\
+    useradd -m -s /bin/bash easybuild && \\
+    chown easybuild:easybuild -R /app && \\
+    chown easybuild:easybuild -R /scratch
+
+USER easybuild
+
+RUN set -x && \\
+    . /usr/share/lmod/lmod/init/sh && \\
+    eb %(eb_opts)s --installpath=/app/ --prefix=/scratch --tmpdir=/scratch/tmp
+
+RUN touch ${HOME}/.profile && \\
+    echo '\\n# Added by easybuild docker packaging' >> ${HOME}/.profile && \\
+    echo 'source /usr/share/lmod/lmod/init/bash' >> ${HOME}/.profile && \\
+    echo 'module use %(init_modulepath)s' >> ${HOME}/.profile && \\
+    echo 'module load %(mod_names)s' >> ${HOME}/.profile
+
+CMD ["/bin/bash", "-l"]
+"""
+
+_DOCKER_TMPLS = {
+    DOCKER_BASE_IMAGE_UBUNTU: DOCKER_UBUNTU1604_TMPL,
+    DOCKER_BASE_IMAGE_CENTOS: DOCKER_CENTOS7_TMPL,
+}
+
+
+class DockerContainer(ContainerGenerator):
+
+    TOOLS = {'docker': None, 'sudo': None}
+
+    RECIPE_FILE_NAME = 'Dockerfile'
+
+    def __init__(self, *args, **kwargs):
+        super(DockerContainer, self).__init__(*args, **kwargs)
+        # NOTE (med): set default value for _container_base
```

Comment:
Is this fixed yet?

Reply:
It's an explanation comment. Nothing should be fixed.